client/services/lessonApi.ts

```diff
@@ -0,0 +1,20 @@
+import { createApi, fetchBaseQuery } from '@reduxjs/toolkit/query/react';
+import { getUserToken } from '@/src/shared/utils';
+
+export const lessonApi = createApi({
+  reducerPath: 'lessonApi',
+  baseQuery: fetchBaseQuery({
+    baseUrl: process.env.NEXT_PUBLIC_API_BASE_URL,
+    prepareHeaders: (headers) => {
+      headers.set('Authorization', `Token ${getUserToken() ?? ''}`);
+      return headers;
+    },
+  }),
+  endpoints: (builder) => ({
+    getLessonByCourseID: builder.query({
+      query: (courseID) => `course/${courseID}/lesson`,
+    }),
```

Comment:
Please update your api for course in BE, and send request to get course not lessons, since the page is all about course ideally, we need to send request to get a course and response should look something like this one below:

```
{
  id:1,
  name: 'course name',
  description: 'course description',
  image: 'preview image link'
  lessons: [
   id: '1',
   title : 'lesson title',
   link: 'link',
   description: 'lesson description',
 ]
}
```

Because the very same data will be used in other tabs, especially settings

Reply:
![image](https://github.com/framgia/sph-lms/assets/115448429/ad1afda2-1ff7-4dac-b502-3007b521135d)